### ISSUE TITLE
dotnet build after deleting “Any CPU” platform from Solution fails

### ISSUE THREAD

**Issue by DvdKhl** [dotnet/sdk#11663]
Hi,

I've got a Solution running with .NET Core 3.1 as the main console application, but it also includes a plain C project.
Up until this [commit](https://github.com/DvdKhl/AVDump3/commit/b74740f72607b1e57f986b64746e795be03420a9#diff-a465f109bb532bec121fadb884852038) building the solution with Visual Studio 2019 or dotnet build worked without any problems.

I'm getting the following error on dotnet build after the commit (Building within VS2019 still works fine):
```
D:\Projects\C#\AVDump3\AVDump3NativeLib\AVDump3NativeLib.vcxproj : warning NU1503: Skipping restore for project 'D:\Projects\C#\AVDump3\AVDump3NativeLib\AVDump3NativeLib.vcxproj'. The project file may be invalid or missing targets required for restore. [D:\Projects\C#\AVDump3\AVDump3.sln]
D:\Projects\C#\AVDump3\AVDump3NativeLib\AVDump3NativeLib.vcxproj(22,3): error MSB4019: The imported project "D:\Microsoft.Cpp.Default.props" was not found. Confirm that the expression in the Import declaration "\Microsoft.Cpp.Default.props" is correct, and that the file exists on disk.
D:\Projects\C#\AVDump3\AVDump3NativeLib\AVDump3NativeLib.vcxproj(22,3): error MSB4019: The imported project "D:\Microsoft.Cpp.Default.props" was not found. Confirm that the expression in the Import declaration "\Microsoft.Cpp.Default.props" is correct, and that the file exists on disk.
```
The only relevant thing that changed is that I removed the "Any CPU" configuration via the IDE from the solution. As soon as I include the "Any CPU" configuration, it works again.

I'm using the following command:
`dotnet build --configuration GithubWorkflow`

This is also reproducible within my Github Workflow: [Success (PreCommit)](https://github.com/DvdKhl/AVDump3/runs/672594715?check_suite_focus=true) / [Error (PostCommit)](https://github.com/DvdKhl/AVDump3/runs/672643901?check_suite_focus=true)

**Comment by sfoslund** [comment]
It sounds like this is a command line only failure, it works in Visual Studio. @wli3 I seem to remember c++ is a Visual Studio only scenario, is that correct?

**Comment by wli3** [comment]
Correct. C++/CLI does not have command line (via dotnet build) support

**Comment by DvdKhl** [comment]
I know that dotnet build cannot build the C Project, but that is not the issue.
Before this [commit](https://github.com/DvdKhl/AVDump3/commit/b74740f72607b1e57f986b64746e795be03420a9#diff-a465f109bb532bec121fadb884852038) dotnet build just ignored the C Project, after the commit it fails.
I was kind of relying on skipping that behavior.

If you don't plan to address this, I'd like you to at least acknowledge the building behavior is inconsistent. If I didn't know what exactly changed to cause this issue, I would have never guessed that missing the "Any CPU" configuration will cause this issue.

**Comment by ZeljkoTodor** [comment]
Hi all. Sorry for asking on this thread, but I stumbled across the same issue with C++ CLI.

I am converting .net 4.8 solution to .net5.0. In this solution most of the projects are csproj, and I have few of vcxproj (C++/CLI).  One CS project is sort of SDK and that one is used by other CS projects, as well as in C++ CLI project. And, the C++ CLI is referenced by another CS project...

@wli3, could you please tell me what would be expected procedure in this case?
Do I still have just one solution, or I need to have few of them?

Thanks
